src/main/java/eu/darkbot/popcorn/def/AntiPush.java

```diff
@@ -45,15 +65,68 @@ public void setConfig(Config config) {
 
     @Override
     public void tick() {
+        tickDrawFire();
+        tickDeathPause();
+    }
+
+    // updating deathStats on tickStopped() because behaviour not ticked when ship isDead()
+    @Override
+    public void tickStopped() {
+        if (config.DEATH_PAUSE_TIME == 0) return;
+
+        updateDeathStats();
+        if (repairManager.isDead()) hasCountedDeath = false;
+        else {
+            Ship killer = ships.stream()
+                    .filter(s -> s.playerInfo.username.equals(repairManager.getKillerName()))
+                    .findFirst()
+                    .orElse(null);
+            if (!hasCountedDeath && killer != null) {
+                deathStats.computeIfAbsent(killer.id, l -> new ArrayList<>())
+                        .add(Instant.now());
+            }
+            hasCountedDeath = true;
+        }
```

Comment:
```suggestion
        boolean isDead = repairManager.isDead();
        if (isDead && !wasDead) {
            ships.stream()
                    .filter(s -> s.playerInfo.username.equals(repairManager.getKillerName()))
                    .findFirst()
                    .ifPresent(killer -> deathStats.computeIfAbsent(
                                    killer.id, i -> new ArrayList<>()).add(Instant.now()));
        }
        wasDead = isDead;
```

Simply save a "wasDead" variable that says if you were dead last tick (rename hasCountedDeath)